### ISSUE TITLE
MEME (meme_meme) tool has dependency issues at http://usegalaxy.org

### ISSUE THREAD

**Issue by jennaj** [galaxyproject/galaxy#1913]
Was meme_meme ready to be deployed on Main or was the install incomplete? I see some activity in tools-iuc for it, but not sure if those mods impact usability this tool version. Error message suggests not and missing dependencies are the problem. meme_fimo works fine.

ping @davebx @gregvonkuster 

**Galaxy Tool ID:**
toolshed.g2.bx.psu.edu/repos/iuc/meme_meme/meme_meme/4.11.0.0

**Error:**
> Tool execution generated the following error message:
> mv: cannot stat `/galaxy-repl/main/jobdir/012/087/12087038/dataset_14747469_files/meme.html': No such file or directory
> The tool produced the following additional output:
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Can't locate XML/Parser/Expat.pm in @INC (@INC contains: /galaxy-repl/main/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/bin /galaxy/main/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/lib/perl /galaxy/main/perl5/lib/x86_64-linux-thread-multi /galaxy/main/perl5/lib /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at /galaxy/main/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/lib/perl/CheckingSAX.pm line 12.
> BEGIN failed--compilation aborted at /galaxy/main/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/lib/perl/CheckingSAX.pm line 12.
> Compilation failed in require at (eval 12) line 3.
> 	...propagated at /usr/share/perl5/base.pm line 94.
> BEGIN failed--compilation aborted at /galaxy/main/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/lib/perl/MemeSAX.pm line 3.
> Compilation failed in require at /galaxy/main/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/bin/meme_xml_to_html line 48.
> Warning: meme_xml_to_html exited abnormally and may have failed to create HTML output.
> Error running MEME.

**Comment by mvdbeek** [comment]
The latest revision of the meme package is `738dc831aeac` and includes expat, so an update should do it.
`

**Comment by jennaj** [comment]
@mvdbeek Thanks Marius. 

@davebx (or whoever does the update) When on main lmk and I'll retest, tx!

**Comment by jennaj** [comment]
@natefoo ping for tool update please

**Comment by jennaj** [comment]
@natefoo @davebx 

It looks like this tool was updated, but it needs some tuning. 

Version on main: MEME - Multiple Em for Motif Elicitation (Galaxy Version 4.11.0.1)

Error: 

Tool execution generated the following error message:
```mv: cannot stat `/galaxy-repl/main/jobdir/XXX/XXX/XXXXXX/dataset_XXX_files/meme.html': No such file or directory```

The tool produced the following additional output:
```
Dataset too large (> 100000).  Rerun with larger -maxsize.
Error running MEME.
```

**Comment by jennaj** [comment]
The updated tool now produces a new error that appears to be dependency related.

> stderr
> 
> mv: cannot stat `/galaxy-repl/main/jobdir/XXX/XXX/XXX/dataset_XXX_files/meme.html': No such file or directory
> 
> stdout
> 
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Can't locate XML/Parser/Expat.pm in @INC (@INC contains: /cvmfs/main.galaxyproject.org/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/bin /galaxy/main/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/lib/perl /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at /galaxy/main/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/lib/perl/CheckingSAX.pm line 12.
> BEGIN failed--compilation aborted at /galaxy/main/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/lib/perl/CheckingSAX.pm line 12.
> Compilation failed in require at (eval 12) line 3.
> 	...propagated at /usr/share/perl5/base.pm line 94.
> BEGIN failed--compilation aborted at /galaxy/main/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/lib/perl/MemeSAX.pm line 3.
> Compilation failed in require at /galaxy/main/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/bin/meme_xml_to_html line 48.
> Warning: meme_xml_to_html exited abnormally and may have failed to create HTML output.
> Error running MEME.

**Comment by gregvonkuster** [comment]
I encountered this same issue on my production ChIP-exo server.  In my case, the ImageMagick-devel package was not available on the system at the time the Galaxy install process compiled ImageMagick as the dependency of meme.  My fix was to uninstall meme, install the ImageMagick-devel RPM, then re-install meme.

**Comment by jennaj** [comment]
thanks @gregvonkuster 

I went back and saw that this error was one that appeared during the intial release of the tools to main, then it went away, and now is back.

Hopefully we can get this update with proper config soon :) 

Q: @davebx would this your area or @natefoo 's ?

**Comment by natefoo** [comment]
This tool depends on graphicsmagick so hopefully the system installation of ImageMagick shouldn't be required. The missing perl package (XML::Parser::Expat) is also a dependency. It looks like this tool has not been updated on Main (the latest version installed on Main is `62d1fae3b7d3`).

**Comment by gregvonkuster** [comment]
Yes, the latest version (4.11.1) depends on graphicsmagick, but version 4.11.0 and earlier depend on imagemagick.  So updating to 4.11.1 may do the trick.

**Comment by natefoo** [comment]
@gregvonkuster I believe there's an error in the `package_meme_4_11_1` tool_dependencies.xml currently on the TS and possibly the TS metadata:

The [TS repo for package_meme_4_11_1](https://toolshed.g2.bx.psu.edu/view/iuc/package_meme_4_11_1/336b41a99b8f) shows:

Repository dependencies - installation of these additional repositories is required
Repository package_libxslt_1_1_28 revision **18e2e2ef0f08** owned by iuc (prior install required)

18e2e2ef0f08 is revision 0 of `package_libxslt_1_1_28`.

However, the [tool_dependencies.xml](https://toolshed.g2.bx.psu.edu/repository/browse_repository?id=5ce038c6e1d98e98) refers to changeset **d63425547b4c** (revision 1).

Unfortunately, even that changeset is not what we want, because 1:63425547b4c was broken (it referred to itself as a dependency when it meant to refer to libxml2. The one we want is the tip, [2:18e2e2ef0f08](https://toolshed.g2.bx.psu.edu/repository/view_changelog?id=db2edea29d6dc127)

Can this be fixed by reuploading `package_meme_4_11_1` to the toolshed, to regenerate the changeset IDs in tool_dependencies.xml (and resetting metadata on the repository if necessary)?

**Comment by natefoo** [comment]
@bgruening updated the repo for me so I think this should be fixed. And just for the record, I had a mistake in my comment above - the `package_meme_4_11_1` dependency list (in the TS ui) referenced 18e2e2ef0f08, which was revision 2.

**Comment by gregvonkuster** [comment]
@natefoo Yes, was just commenting on your comment above regarding the tip of package_libxslt_1_1_28, so glad you caught it before I did.  So am I understanding correctly that this issue is now corrected for meme?

**Comment by natefoo** [comment]
Thanks @gregvonkuster. Just a bit further now: `perl_xml_parser_expat` fails to install because it depends on expat, which is not installed or depended on. There is a `package_expat_2_1` IUC package in the TS which might work, however.

**Comment by gregvonkuster** [comment]
So that means that this recipe is not working?  https://github.com/galaxyproject/tools-iuc/tree/master/packages/package_perl_xml_parser_expat_2_41  How has this issue been missed so far?  Its a fairly old recipe.

**Comment by natefoo** [comment]
I believe it'll work on anything that already has the C expat development package installed (i.e. `expat-devel` on RHELish), but fail otherwise.

**Comment by gregvonkuster** [comment]
Hopefully this https://github.com/galaxyproject/tools-iuc/pull/917 will correct it?

**Comment by jennaj** [comment]
Not sure of repo/main update status, but the tool is still failing at http://usegalaxy.org. This is an example current error message: 


> stderr
> 
> mv: cannot stat `/galaxy-repl/main/jobdir/XXX/XXX/XXX/dataset_XXX_files/meme.html': No such file or directory
> 
> stdout
> 
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Warning: Can not convert EPS file to PNG as no install of Image Magick or Ghostscript is usable.
> Can't locate XML/Parser/Expat.pm in @INC (@INC contains: /cvmfs/main.galaxyproject.org/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/bin /galaxy/main/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/lib/perl /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at /galaxy/main/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/lib/perl/CheckingSAX.pm line 12.
> BEGIN failed--compilation aborted at /galaxy/main/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/lib/perl/CheckingSAX.pm line 12.
> Compilation failed in require at (eval 12) line 3.
> 	...propagated at /usr/share/perl5/base.pm line 94.
> BEGIN failed--compilation aborted at /galaxy/main/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/lib/perl/MemeSAX.pm line 3.
> Compilation failed in require at /galaxy/main/deps/meme/4.11.0/iuc/package_meme_4_11_0/6ee2e1225125/bin/meme_xml_to_html line 48.
> Warning: meme_xml_to_html exited abnormally and may have failed to create HTML output.
> Error running MEME.

**Comment by jennaj** [comment]
Any updates on when this will go to Main? @natefoo @gregvonkuster

**Comment by natefoo** [comment]
@jennaj Unfortunately it's not fixed, so there's nothing to push to Main.

**Comment by jennaj** [comment]
@natefoo Thanks, I appreciate the update! Ping @galaxyproject/guac when ready and one of us will test

**Comment by jennaj** [comment]
Still a problem. Current plans to get fix on Main?

**Comment by nekrut** [comment]
what kind of problem?

**Comment by jennaj** [comment]
The tool was failing. I still need to do my test. Others testing this morning I think and it was a pass. When I'm done, and if passes, will close out the ticket.
